api/rest/restapi.go

```diff
@@ -352,6 +363,56 @@ func (api *API) graphHandler(w http.ResponseWriter, r *http.Request) {
 	sendResponse(w, err, graph)
 }
 
+func (api *API) addFileHandler(w http.ResponseWriter, r *http.Request) {
+	contentType := r.Header.Get("Content-Type")
+	mediatype, _, _ := mime.ParseMediaType(contentType)
+	var f files.File
+	if mediatype == "multipart/form-data" {
+		reader, err := r.MultipartReader()
+		if err != nil {
+			fmt.Printf("The error getting a multipartreader: %s", err.Error())
+			sendAcceptedResponse(w, nil)
+			return
+		}
+
+		f = &files.MultipartFile{
+			Mediatype: mediatype,
+			Reader:    reader,
+		}
+
+	} else {
+		fmt.Printf("the mediatype: %s", mediatype)
+		sendAcceptedResponse(w, nil)
```

Comment:
mm is this a placeholder for non-streaming?

Reply:
No, thanks for pointing this out.  I should be returning an error here.  Right now the file handler requires that files are transmitted as multipart/form mediatype (we could probably be a little less restrictive in the future).